### PR TITLE
feat: included network_id as part of the primary key for fare_leg_rules.txt

### LINKED ISSUE #1889: Modify primary key for fare_leg_rules.txt to include network_id
### Describe the problem

There are false positives in the validator for `duplicate_key` [(see acceptance tests of new errors)](https://github.com/MobilityData/gtfs-validator/pull/1804#issuecomment-2313617185), because `network_id` is not a part of the primary key for fare_leg_rules.txt as it should be according to the spec. 

### Proposed solution

Make sure that `network_id` is included as part of the primary key for fare_leg_rules.txt. 

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**
Closes #1889 
[gtrans-ca-us.zip](https://github.com/user-attachments/files/17415663/gtrans-ca-us.zip)

**Expected behavior:** 
test dataset: 
Before:
image

After:
image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/test/java/org/mobilitydata/gtfsvalidator/table/GtfsFareLegRuleDuplicateKeyTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        .setFromAreaId(fromAreaId)
+        .setToAreaId(toAreaId)
+        .setFromTimeframeGroupId(fromTimeframeGroupId)
+        .setToTimeframeGroupId(toTimeframeGroupId)
+        .setFareProductId(fareProductId)
+        .build();
+  }
+
+  @Test
+  public void noDuplicateKeyNoticeWithNetworkIdAsPrimaryKey() {
```
Comment (jcpitre) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/table/GtfsFareLegRuleDuplicateKeyTest.java:32]: Maybe you could add a test proving that a duplicate key notice is raised if network is the same?

**Comment by emmambd** [comment]
LGTM!
